### PR TITLE
EOS-19806: Need factory reset support through mini provisioner interface

### PR DESCRIPTION
**Pull request by gargpalak048**
## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Need factory reset support through mini provisioner interface
  </code>
</pre>
## Solution
<pre>
  <code>
    Added support for Reset Interface for SSPL component
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - [ ] Yes
    - [ ] No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
  </code>
</pre>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: low-level/files/opt/seagate/sspl/setup/sspl_reset.py [hunk 1]
```diff
@@ -24,67 +24,32 @@ def del_file(self, filename):
...
-        # Remove sspl_conf
-        self.del_file(file_store_config_path)
-
-        # Remove sspl-configured file
-        self.del_file(SSPL_CONFIGURED)
+        # Stop SSPL service if state is active
+        service_state = dbus_service.get_state('sspl-ll.service')
+        if service_state._state == 'active':
+            logger.info(f'For {self.name}, SSPL service should be stopped'
+            'through HA interfaces')
```
Comment (thavanathan) [inline low-level/files/opt/seagate/sspl/setup/sspl_reset.py:39]: continuation line under-indented for visual indent. Please make sure to run flake8 before submitting PR.

Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_reset.py]: Fixed flake8 issues.

**Comment by gargpalak048** [comment]
> low-level/files/opt/seagate/sspl/setup/sspl_setup.py#L384
1 blank line required before class docstring (found 0) (D203)

Blank line already present. This is a invalid codacy comment

**Comment by cdeshmukh** [comment]
test this please

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 2]
```diff
@@ -369,14 +380,13 @@ def process(self):
 
 
 class ResetCmd(Cmd):
-    """Performs SSPL config reset. Options: hard, soft.
-    'hard' is used to reset configs and clean log directory where
-    'soft' is to clean only the data path.
+
+    """Performs SSPL config reset.
+    Reset Interface is used to reset configs and clean log directory
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:385]: Config is not touched in reset now right.

>Reset Interface is used to reset meta data and clean log directory

File: low-level/files/opt/seagate/sspl/setup/sspl_reset.py [hunk 3]
```diff
@@ -2,89 +2,56 @@
...
-
-        # Remove sspl_conf
-        self.del_file(file_store_config_path)
-
-        # Remove sspl-configured file
-        self.del_file(SSPL_CONFIGURED)
+        # Stop SSPL service if state is active
+        service_state = dbus_service.get_state('sspl-ll.service')
+        if service_state._state == 'active':
+            logger.info(f'For {self.name}, SSPL service should be stopped'
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_reset.py:36]: logger.warning ("SSPL service should have been stopped before Factory reset is invoked, stopping now")

File: low-level/files/opt/seagate/sspl/setup/sspl_reset.py [hunk 4]
```diff
@@ -2,89 +2,56 @@
...
                                            SSPL_CONFIG_INDEX,
                                            sspl_config_path)
-from cortx.utils.process import SimpleProcess
 from cortx.utils.service import DbusServiceHandler
 from cortx.utils.conf_store import Conf
-from files.opt.seagate.sspl.setup.setup_error import SetupError
 from files.opt.seagate.sspl.setup.setup_logger import logger
 
+
 class HardReset:
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_reset.py:13]: HardReset needs to be renamed Reset alone, even though actions are related to hard reset. as no more hard/soft reset concept applied now in R2

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py (hunk 2 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_setup.py]: Done the change

File: low-level/files/opt/seagate/sspl/setup/sspl_reset.py (hunk 3 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_reset.py]: Done

File: low-level/files/opt/seagate/sspl/setup/sspl_reset.py (hunk 4 above)
Comment (gargpalak048) [reply low-level/files/opt/seagate/sspl/setup/sspl_reset.py]: Ack.
Class name changed from HardReset to Reset.

**Review by sandeepsgt — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
test this please

File: low-level/files/opt/seagate/sspl/setup/sspl_setup.py [hunk 5]
```diff
@@ -371,14 +382,12 @@ def process(self):
 
 
 class ResetCmd(Cmd):
-    """Performs SSPL config reset. Options: hard, soft.
-    'hard' is used to reset configs and clean log directory where
-    'soft' is to clean only the data path.
+    """Reset Interface is used to reset Data/MetaData
+    and clean log files
```
Comment (cdeshmukh) [inline low-level/files/opt/seagate/sspl/setup/sspl_setup.py:386]: D211 is conflicting with D203, ignoring D211

**Comment by cdeshmukh** [comment]
Only codacy issue seen is D211, which is conflicting with D203, ignoring D211

**Comment by cdeshmukh** [comment]
PR VM deployment wont be possible today

[6:02 PM] Priyank Dalal
    @All -- we disabled the cleanup job for  single node VM to reduce the load on Infra and allowing 3node to go through. 
Today, you may  not get VM for PR's. Please use your own VM.

**Comment by thavanathan** [comment]
test this please

**Comment by thavanathan** [comment]
@gargpalak048 please fix Codacy issues.

**Comment by gargpalak048** [comment]
> @gargpalak048 please fix Codacy issues.

@thavanathan only D211 issue is seen, which is conflicting with D203, we can ignore that.

**Comment by cdeshmukh** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/SSPL/275/
build got aborted, with a timeout issue 


>[Checks API] No suitable checks publisher found.
Setting status of ab2179cbe8fa1a521ca82ceebf3d0f3e88eaca4d to FAILURE with url http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Cortx-Deployment/job/SSPL/275/ and message: 'Build finished. '
Using context: VM Deployment
Timeout has been exceeded
Finished: ABORTED

**Comment by cdeshmukh** [comment]
@gowthamchinna @priyankdal VM unavailability are really blocking PRs from days now, this is one such example

**Pull request merged**